tests/outputs/TestDiagram.cpp

```diff
@@ -114,43 +130,37 @@ TEST(Diagram, writeWithCurveAndDefaultPoints) {
                           3., 30., 33., 330.),
       GeneratorDefinition("G3", GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN, "03", {}, 4., 40., 44., 440.)};
 
-  outputPath.append(filename);
-  dfl::outputs::Diagram DiagramWriter(dfl::outputs::Diagram::DiagramDefinition(basename, outputPath.generic_string(), generators));
+  dfl::outputs::Diagram DiagramWriter(dfl::outputs::Diagram::DiagramDefinition(basename, outputDirectory.generic_string(), generators));
 
   DiagramWriter.write();
-
-  boost::filesystem::path reference("reference");
-  reference.append(basename);
-  reference.append(filename);
-
-  dfl::test::checkFilesEqual(outputPath.generic_string(), reference.generic_string());
+  testMultiplesFilesEquality(generators, outputDirectory, basename, prefixDir);
 }
 
 TEST(Diagram, writeEmpty) {
   using dfl::algo::GeneratorDefinition;
 
   std::string basename = "TestDiagram";
-  std::string filename = "TestDiagramEmpty.txt";
-  boost::filesystem::path outputPath("results");
-  outputPath.append(basename);
+  std::string prefixDir = "Empty";
+  boost::filesystem::path outputDirectory("results");
+  outputDirectory.append(basename);
 
-  if (!boost::filesystem::exists(outputPath)) {
-    boost::filesystem::create_directories(outputPath);
+  if (!boost::filesystem::exists(outputDirectory)) {
+    boost::filesystem::create_directories(outputDirectory);
   }
+  outputDirectory.append(prefixDir + dfl::outputs::constants::diagramDirectorySuffix);
```

Comment:
Please make sure the directory is empty

Reply:
Do you mean I have to test twice the fact that it is empty ? It is already done 
` std::ifstream stream(directoryPath);
    ASSERT_TRUE(stream.fail()) << "The Diagram directory " << directoryPath << " has been created to even though there are no generators which model requires it."
                               << std::endl; ` 
here

Reply:
ok!